WTG.Analyzers.Test/TestData/BooleanLiteralCombiningAnalyzer/AndOrAssignment/Result.cs

```diff
@@ -0,0 +1,25 @@
+using System.Diagnostics.Contracts;
+
+public static class Bob
+{
+	public static bool Method1(bool a)
+	{
+		a = true;
+		return a;
+	}
+	public static bool Method2(bool a)
+	{
+		a;
```

Comment:
So the goal here is not to remove the expression, but to transform it into a compiler error?

Reply:
Not exactly. the main concern is not removing something that may have side effects. In this particular case, it's safe, but it could just as easily have been:

```
Foo().a |= true;
```

I think we can still do better (without having to resort to using the `SemanticModel`), but thought I would leave that for another time as it will affect other consumers of the `ExpressionRemover` and the same improvement should be applied to other existing cases where we produce 'awkward' results in order to avoid removing side-effects.